### DISCUSSION TITLE
Help babel/babel with their Renovate config

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#10381] [Request Help] [answered]
I posted this comment on the Renovate Dashboard of the `babel/babel` repository:

> Did you know you can update lockfiles (`yarn.lock`) with Renovate bot? That might be nice to do on a weekly schedule.
> This option is called `lockfileMaintenance`, if you turn this on, the default behavior is for it to run on Monday each week.
> You can then review the lockfile update and merge if happy with it.
> 
> https://docs.renovatebot.com/configuration-options/#lockfilemaintenance
> 
> Optional: you could even tell Renovate bot to automerge the lockfile update if test pass. That way you don't have to do anything. But this takes a bit of work to setup properly.

One of the members of Babel (@nicolo-ribaudo) replied:

> @HonkingGoose Thanks for the information!
> 
> Can we tell Renovate to update some deps in that same lockfile PR? Ideally I would like to replace @babel-bot with Renovate for https://github.com/babel/babel/pull/12968 and do both in the same PR.

---

It appears to me they want two things (at least) from Renovate bot:

1. Update dev/prod dependencies in the same PR as the lockfile
2. Use the commit hash of a file in a makefile (https://github.com/babel/babel/pull/12968/files)

I don't think we can update anything but the lockfile in the lockfile update PRs? Maybe we can help by grouping certain dependencies together...

I don't know enough about regex and fetching git commits to be able to help with number 2. Maybe we could use something like the "update references config" that updates references to a Dockerfile in our documentation? So something like this update PR https://github.com/renovatebot/renovate/pull/8757, but we'd need a way to "grab the commit hash from elsewhere".

Renovate config for Regex manager: https://github.com/renovatebot/.github/blob/930726ea9db00b581216bc7618a832dfcd81155a/default.json#L96-L105

I hope you don't mind me trying to get config help for Babel. 😉 🙈

**Comment by rarkins** [comment]
> Update dev/prod dependencies in the same PR as the lockfile

By this you/they mean "lock file maintenance"? And in practice they mean they want a PR which does both:
- updates dependencies in `package.json`
- fully refreshes/recreates the lock file every time?

**Reply by nicolo-ribaudo** [reply]
We would like to both update the lockfile and the pinned commit hash in the same PR, since those are the two things that may require us to regenerate our fixtures.

This is more a "nice to have" than a requirement.

**Reply by rarkins** [reply]
I think I misunderstood originally. You mean the `test262` commit hash?

**Comment by rarkins** [comment, marked as answer]
I think (2) is probably doable too with regex manager, but I don't fully understand what they're doing. Which repo is it a commit hash from?

**Reply by nicolo-ribaudo** [reply]
It's a commit from https://github.com/tc39/test262. We currently update it weekly, but only if there has been any update.

**Reply by rarkins** [reply]
If it were to be automated, would you want it updated right away (after a change is detected), or just checked once a week?

**Reply by nicolo-ribaudo** [reply]
Just once a week: that repository is only used in our tests, so it's ok to have a slightly out-of-date version.

**Reply by rarkins** [reply]
This should work:

```json
{
  "regexManagers": [
    {
      "fileMatch": ["^Makefile$"],
      "matchStrings": ["TEST262_COMMIT = (?<currentDigest>.*)\\n"],
      "datasourceTemplate": "git-refs",
      "depNameTemplate": "test262",
      "lookupNameTemplate": "https://github.com/tc39/test262.git",
      "currentValueTemplate": "main"
    }
  ],
  "packageRules": [
    {
      "matchPackageNames": ["test262"],
      "schedule": [
        "before 4am on Monday"
      ]
    }
  ]
}
```

Tested here: https://github.com/renovate-tests/babel1/pull/3

**Reply by nicolo-ribaudo** [reply]
Thanks! :heart:
